### ISSUE TITLE
Upgrade to or support building with Qt 6

### ISSUE THREAD

**Issue by JamieWhite** [Xaymar/obs-StreamFX#811]
### Explain the Feature/Enhancement

OBS currently supports Qt 5 and Qt 6 on their repo. They're aiming to get a beta Qt 6 build out in the next month and a final release in August. Plugins built against Qt 5 will not work with OBS Qt 6 so they'll either need to move to Qt 6 or support both. More details here https://github.com/obsproject/obs-studio/discussions/6481

### What makes this feature/enhancement useful for StreamFX and its users?

Mainly future compatibility with OBS.

Qt 6's main benefit is it supports Apples M1 chips natively. In Qt 5 OBS was running via rosetta which has a fairly significant performance impact. 

### Checklist

- [ ] I am offering a monetary bounty/reward for the implementation of this Feature/Enhancement.
- [X] There is no other way to currently do this in StreamFX or OBS Studio.

**Comment by Xaymar** [comment]
Since StreamFX will always require a recent version of OBS Studio, only supporting one will be enough. It technically already supports Qt6 if built with it, as the new CMake does download Qt via obs-dependecies

**Comment by JamieWhite** [comment]
Interesting. I couldn't get it to build and when i looked around i saw a bunch of Qt5 references so assumed it was that. Im admittedly not much of a C++ programmer so apologies if this is really obvious.

I've just tried building it again and this is the error Im getting when following your steps https://github.com/Xaymar/obs-StreamFX/wiki/Building#building-bundled-with-obs-studio and using the OBS mac instructions here https://github.com/obsproject/obs-studio/wiki/Build-Instructions-For-Mac:


```
-- StreamFX: Added post-build step for adjusting libobs linking path.
-- StreamFX: Added post-build step for adjusting libobs-frontend-api linking path.
-- StreamFX: Added post-build step for adjusting Qt5::Core linking path (Found: /Users/jamie/Documents/Coding/obs-build-dependencies/obs-deps/lib/cmake/Qt5 resolved to @rpath/QtCore.framework/Versions/5/QtCore;/Users/jamie/Documents/Coding/obs-build-dependencies/obs-deps/lib/QtCore.framework/QtCore (architecture arm64):;@rpath/QtCore.framework/Versions/5/QtCore).
-- StreamFX: Added post-build step for adjusting Qt5::Gui linking path (Found: /Users/jamie/Documents/Coding/obs-build-dependencies/obs-deps/lib/cmake/Qt5 resolved to @rpath/QtGui.framework/Versions/5/QtGui).
-- StreamFX: Added post-build step for adjusting Qt5::Widgets linking path (Found: /Users/jamie/Documents/Coding/obs-build-dependencies/obs-deps/lib/cmake/Qt5 resolved to @rpath/QtWidgets.framework/Versions/5/QtWidgets).
CMake Deprecation Warning at cmake/Modules/ObsHelpers.cmake:374 (message):
  OBS: The install_obs_plugin_with_data command is deprecated and will be
  removed soon.  Use 'setup_plugin_target' instead.
Call Stack (most recent call first):
  cmake/Modules/DeprecationHelpers.cmake:148 (obs_status)
  UI/frontend-plugins/streamfx/CMakeLists.txt:2104 (install_obs_plugin_with_data)


-- OBS:  ENABLED    StreamFX
-- Configuring done
CMake Error at UI/frontend-plugins/streamfx/cmake/clang/Clang.cmake:235 (file):
  Error evaluating generator expression:

    $<COMPILE_LANG_AND_ID:CXX,AppleClang,Clang>

  $<COMPILE_LANG_AND_ID:lang,id> may only be used with binary targets to
  specify include directories, compile definitions, and compile options.  It
  may not be used with the add_custom_command, add_custom_target, or
  file(GENERATE) commands.
Call Stack (most recent call first):
  UI/frontend-plugins/streamfx/CMakeLists.txt:2026 (generate_compile_commands_json)


CMake Error at UI/frontend-plugins/streamfx/cmake/clang/Clang.cmake:235 (file):
  Error evaluating generator expression:

    $<COMPILE_LANG_AND_ID:CXX,AppleClang,Clang>

  $<COMPILE_LANG_AND_ID:lang,id> may only be used with binary targets to
  specify include directories, compile definitions, and compile options.  It
  may not be used with the add_custom_command, add_custom_target, or
  file(GENERATE) commands.
Call Stack (most recent call first):
  UI/frontend-plugins/streamfx/CMakeLists.txt:2026 (generate_compile_commands_json)


-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.
  + ERROR during build step: configure build
```

**Comment by Xaymar** [comment]
In-tree builds are for development, so you need to turn off ENABLE_CLANG. See other issue with identical error: https://github.com/Xaymar/obs-StreamFX/issues/799

**Comment by Xaymar** [comment]
Actual Qt6 support in StreamFX will be available when OBS Studio 28.0 is actually out as stable, and no earlier. In-development changes are impossible to verify stability against, and will just cause more work than I want to do - but the PR button is available if someone wants to add Qt6 detection and support early.
